sklearn/tests/test_naive_bayes.py

```diff
@@ -868,6 +868,25 @@ def test_alpha():
                          X, y, classes=[0, 1])
 
 
+def test_check_alpha():
+    # Test force_alpha if alpha < _ALPHA_MIN
+    _ALPHA_MIN = 1e-10  # const
+    msg1 = ('alpha too small will result in numeric errors, '
+            'force_alpha was set to True, '
+            'proceeding without changing alpha.')
+    msg2 = ('alpha too small will result in numeric errors, '
+            'setting alpha = %.1e' % _ALPHA_MIN)
+    b = BernoulliNB(alpha=0, force_alpha=True)
+    with pytest.warns(UserWarning, match=msg1):
+        assert b._check_alpha() == 0
+    b = BernoulliNB(alpha=0, force_alpha=False)
+    with pytest.warns(UserWarning, match=msg2):
+        assert b._check_alpha() == _ALPHA_MIN
+    b = BernoulliNB(alpha=0)
+    with pytest.warns(UserWarning, match=msg2):
+        assert b._check_alpha() == _ALPHA_MIN
```

Comment:
You can use `assert_warns_message` to assert that warnings are thrown in those cases. 🙂

Reply:
i'm following #19414 to not use assert_warns_message - this will allow me to use regex for the warning message

Reply:
Indeed, this is a better practice :+1: